### PR TITLE
Remove the obsolete parameter "use_dwa"

### LINKED ISSUE #1110: Can we remove this use_dwa parameter?
Can we remove this use_dwa parameter? Not part of this PR, but if so, can we file an issue and do it soon? It seems silly to keep something that was deprecated in ROS.

_Originally posted by @mkhansen-intel in https://github.com/ros-planning/navigation2/pull/1045_

### PR DESCRIPTION
**Pull request by crdelsey**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | fixes #1110 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | TB3 in Gazebo|

---

## Description of contribution in a few bullet points

* Removes the use_dwa parameter

### CODE REVIEW

File: nav2_dwb_controller/dwb_plugins/src/standard_traj_generator.cpp [hunk 1]
```diff
@@ -88,18 +87,6 @@ void StandardTrajectoryGenerator::initializeIterator(
   velocity_iterator_->initialize(nh, kinematics_);
 }
 
-void StandardTrajectoryGenerator::checkUseDwaParam(
```
Comment (SteveMacenski) [inline nav2_dwb_controller/dwb_plugins/src/standard_traj_generator.cpp]: Please add to the DWB readme that they should use the `LimitedAccelGenerator` for DWA behavior and `StandardTrajectoryGenerator` for DWB behavior

**Comment by SteveMacenski** [comment]
@crdelsey this PR is starting to turn into something else... 

Can you just add the `Note:` in the readme about which generator to use for DWA-like behavior and we can merge this?

**Comment by crdelsey** [comment]
> Can you just add the Note: in the readme about which generator to use for DWA-like behavior and we can merge this?

There is no good place to put that in the readme, IMHO. To me that would make the readme worse. I  could pull the docs changes out of this PR and just merge the code, and do the docs as a separate PR.

**Comment by SteveMacenski** [comment]
I'm totally OK with removing the parameter and making the choice of plugin based on your desire... but there has to be a note somewhere in DWB that hey, this other option is "here solely for DWA-like behavior". Readme, inline code, in plugin loader code, somewhere. This can't be tribal knowledge lost to history.

**Comment by crdelsey** [comment]
@SteveMacenski Are you ok with the doc changes I added here?

**Pull request merged**